[Term entry] Kotlin .drop last() (#3898)
Codecademy/docs: content/kotlin/concepts/strings/terms/dropLast/dropLast.md

---
Title: 'dropLast()'
Description: 'Returns a string with n characters removed.'
Subject:
  - 'Computer Science'
Tags:
  - 'Methods'
  - 'Data Types'
  - 'Characters'
CatalogContent:
  - 'learn-kotlin'
  - 'paths/computer-science'
---

> The Kotlin method, **`.dropLast()`** returns a string with the last `(n)` characters removed.

Reviewer: ```suggestion
The Kotlin method, **`.dropLast()`** returns a string with the last `n` characters removed.
```
There's no need to add brackets.

## Syntax

```pseudo
value string = "Example String"

> println(string.dropLast())

Reviewer: This seems more like an example. You can just add a valid syntax, you can refer to the official documentation too but don't copy and paste it directly as it will be considered Plagiarism.

```

### Special Cases

> `IllegalArguementException` if `n` is negative.

Reviewer: Please try to describe this special case in detail.

## Example

> The example below creates a string, `string`, then uses the `.dropLast()` method to return astring with the last character and a string the last five characters removed.

Reviewer: ```suggestion
The example below creates a string, `string`, then uses the `.dropLast()` method to return a string with the last five characters removed.
```

Author: I'll make the changes, thanks.

```kotlin
value string = "Example String"
println(string.dropLast())

> println(string.dropLast(5))

Reviewer: The code is not executing properly. Please correct the code and remember to write it inside the main function otherwise it'll throw an error.

---
Title: '.dropLast()'
Description: 'Returns a string with the last n characters removed.'
Subject:
  - 'Computer Science'
Tags:
  - 'Methods'
  - 'Data Types'
  - 'Characters'
CatalogContent:
  - 'learn-kotlin'
  - 'paths/computer-science'
---

The Kotlin method, **`.dropLast()`** removes the last character of a string or the last `n` number of characters from a string. This method does not modify the original list, it creates and returns a new list.

## Syntax

```pseudo

> fun String.dropLast(n: Int): String

Reviewer: Add a line explaining the parameter.

---
Title: 'dropLast()'
Description: 'Returns a string with n characters removed.'
Subject:
  - 'Computer Science'
Tags:
  - 'Methods'
  - 'Data Types'
  - 'Characters'
CatalogContent:
  - 'learn-kotlin'
  - 'paths/computer-science'
---

> The Kotlin method, **`.dropLast()`** removes the last character of a string or the last `n` number of charcters from a string. This method does not modify the original list, it creates and returns a new list.

Reviewer: ```suggestion
The Kotlin method, **`.dropLast()`** removes the last character of a string or the last `n` number of characters from a string. This method does not modify the original list, it creates and returns a new list.
```

## Syntax

```pseudo

> string.dropLast()

Reviewer: The syntax is not correct, there needs to be a parameter that is passed into the function.

```

### Special Cases

`IllegalArguementException` error thrown if `n` is a negative number. A negative number of characters can not be removed.

## Example

The example below creates a string, `string`, then uses the `.dropLast()` method to return a string with the last five characters removed.

```kotlin
import kotlin.test.*
fun main(args: Array<Strings> {
val string = "Example String"
println(string.dropLast())
println(string.dropLast(5))

> }

Reviewer: The code has many errors please test it with an online compiler or any code editor.

}

> 

Reviewer: @Kadif-Ravik You forgot to add the backticks after the code, please add it and keep a line space after the code block.

---

> Title: 'dropLast()'

Reviewer: ```suggestion
Title: '.dropLast()'
```

> Description: 'Returns a string with n characters removed.'

Reviewer: ```suggestion
Description: 'Returns a string with the last n characters removed.'
```

Subject:
  - 'Computer Science'
Tags:
  - 'Methods'
  - 'Data Types'
  - 'Characters'
CatalogContent:
  - 'learn-kotlin'
  - 'paths/computer-science'
---

> The Kotlin method, **`.dropLast()`** returns a string with the last `n` characters removed.

Reviewer: Describe the method in 2-3 lines.

## Syntax

```pseudo

> println(string.dropLast())

Reviewer: Write the syntax properly, don't use println(), it seems like a line from a program.

> The Kotlin method, **`.dropLast()`** removes the last character of a string or the last `n` number, a positive integer, of characters from a string. This method does not modify the original list, it creates and returns a new list.

Reviewer 2: ```suggestion
The Kotlin method, **`.dropLast()`** removes one or `n` number of characters from the end of the string. Here, `n` is a positive integer given as a parameter to the method. This method does not modify the original list, it creates and returns a new list.
```

> string.dropLast(2)

Reviewer: ```suggestion
fun String.dropLast(n: Int): String
```
As it is a syntax, there should not be a specific value in the place of a parameter. This looks more of a proper syntax, please have a look at it.

```

- The parameter `n` is the number of characters to be removed from the string.

> ### Special Cases
> 

Reviewer 2: ```suggestion
```
Since we added the "Note" section remove the "Special Cases" heading.

> `IllegalArguementException` error thrown if `n` is a negative number. A negative number of characters can not be removed.

Reviewer 2: ```suggestion
>**Note:** `IllegalArguementException` error thrown if `n` is a negative number. A negative number of characters can not be removed.
```
This seems more appropriate.

The example below creates a string, `string`, then uses the `.dropLast()` method to return a string with the last five characters removed.

```kotlin
import java.util.Locale
import kotlin.test.*
fun main(args: Array<Strings> {
value string = "<<<Example String>>>"
println(string.dropLast())
println(string.dropLast(5))

> }

Reviewer: The code is still not compiling properly. Please make the necessary changes.

> ```kotlin
> import kotlin.test.*
> fun main(args: Array<String>) {
> val string = "Example String"
> println(string.dropLast(1))
> println(string.dropLast(5))
> }
> 
> ```

Reviewer: ```suggestion
```kotlin

fun main() {
  val string = "Example String"
  println(string.dropLast(1))
  println(string.dropLast(5))
}
```
```
There's no need to add the import statement as the method is working without that line. The code should be formatted as per the documentation.

> 
> fun main() {
>   val string = "Example String"
>   println(string.dropLast(1))
>   println(string.dropLast(5))
> }

Reviewer: ```suggestion
fun main() {
  val string = "Example String"
  println(string.dropLast(1))
  println(string.dropLast(5))
}
```
